src/Host.cpp

```diff
@@ -222,19 +222,85 @@ Host::~Host()
 void Host::saveModules(int sync, bool backup)
 {
     QMapIterator<QString, QStringList> it(modulesToWrite);
-    QStringList modulesToSync;
-    QString dirName = mudlet::getMudletPath(mudlet::moduleBackupsPath);
-    QDir savePath = QDir(dirName);
-    if (!savePath.exists()) {
-        savePath.mkpath(dirName);
+    mModulesToSync.clear();
+    QString savePath = mudlet::getMudletPath(mudlet::moduleBackupsPath);
+    auto savePathDir = QDir(savePath);
+    if (!savePathDir.exists()) {
+        savePathDir.mkpath(savePath);
     }
     while (it.hasNext()) {
         it.next();
         QStringList entry = it.value();
+        QString moduleName = it.key();
         QString filename_xml = entry[0];
-        // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#HH-mm-ss" (1 of 6)
-        QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+
+        if (backup) {
+            // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#HH-mm-ss" (1 of 6)
+            QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+            savePathDir.rename(filename_xml, savePath + moduleName + time); //move the old file, use the key (module name) as the file
+        }
+
+        auto writer = new XMLexport(this);
+        writers.insert(filename_xml, writer);
+        writer->writeModuleXML(moduleName, filename_xml);
+
+        if (entry[1].toInt()) {
+            mModulesToSync << moduleName;
+        }
+    }
+    modulesToWrite.clear();
+
+    if (sync) {
+        connect(this, &Host::profileSaveFinished, this, &Host::slot_reloadModules);
+    }
+}
+
+void Host::slot_reloadModules()
+{
+    //synchronize modules across sessions
+    QMap<Host*, TConsole*> activeSessions = mudlet::self()->mConsoleMap;
+    QMapIterator<Host*, TConsole*> sessionIterator(activeSessions);
+    while (sessionIterator.hasNext()) {
+        sessionIterator.next();
+        Host* host = sessionIterator.key();
+        if (host->mHostName == mHostName) {
```

Comment:
Given that `mHostName` is protected by a `QMutex` it is probably not a good idea to poke around in the other instance's members directly - I think you are supposed to use `otherHost->getHostName()` instead...

Reply:
Updated